src/select-menu/src/Option.js

```diff
@@ -37,38 +52,47 @@ const Option = memo(
       textProps = selectedProps
     }
 
+    const { className: themedClassName, ...boxProps } = useStyleConfig(
+      'Option',
+      emptyObject,
+      pseudoSelectors,
+      internalStyles
+    )
+
+    const { getRef, ...listBehaviorProps } = useListBehavior({
+      isSelectable,
+      isSelected,
+      disabled,
+      onSelect
+    })
+
+    const callbackRef = useMergedRef(getRef, forwardedRef)
+
     return (
-      <TableRow
-        isSelectable={isSelectable && !disabled}
-        isHighlighted={isHighlighted}
-        onSelect={onSelect}
-        onDeselect={onDeselect}
-        isSelected={isSelected}
+      <Pane
         style={style}
-        display="flex"
-        alignItems="center"
-        borderBottom="muted"
-        {...(isSelected
-          ? { boxShadow: `inset 2px 0 0 ${tokens.selectedOptionColor}` }
-          : {})}
+        className={themedClassName}
```

Comment:
Given that we previously supported `className` (via `...rest`) we should use `cx(themedClassName, className)` here.

Also note that if a consumer passed in a className right now it would override our themed className!